### PR TITLE
Adds detection for Yandex Browser Lite and improves version detection for Yandex Browser

### PR DESCRIPTION
**Pull request by liviuconcioiu**
Improves detection for Yandex browsers

### CODE REVIEW

File: regexes/client/browsers.yml [hunk 1]
```diff
@@ -515,6 +515,27 @@
...
 
+# Yandex Browser (alpha)
+- regex: 'YaBrowser(?:/(\d+[\.\d]*)) \(alpha\)?'
+  name: 'Yandex Browser (alpha)'
+  version: '$1'
+  engine:
+    default: 'Blink'
+
+# Yandex Browser (beta)
+- regex: 'YaBrowser(?:/(\d+[\.\d]*)) \(beta\)?'
```
Comment (mimmi20) [inline regexes/client/browsers.yml]: Is the last `?` on the right position?

File: regexes/client/browsers.yml [hunk 2]
```diff
@@ -515,6 +515,27 @@
...
+  name: 'Yandex Browser (beta)'
+  version: '$1'
+  engine:
+    default: 'Blink'
+
+# Yandex Browser Lite
+- regex: 'YaBrowser(?:/(\d+[\.\d]*)) \(lite\)?'
```
Comment (mimmi20) [inline regexes/client/browsers.yml]: Is the last `?` on the right position?

File: regexes/client/browsers.yml [hunk 3]
```diff
@@ -515,6 +515,27 @@
   name: 'Baidu Spark'
   version: '$1'
 
+# Yandex Browser (alpha)
+- regex: 'YaBrowser(?:/(\d+[\.\d]*)) \(alpha\)?'
```
Comment (mimmi20) [inline regexes/client/browsers.yml]: Is the last `?` on the right position?

File: regexes/client/browsers.yml (hunk 1 above)
Comment (liviuconcioiu) [reply regexes/client/browsers.yml]: Yes, is based on the regex for Yandex Browser.

File: regexes/client/browsers.yml (hunk 3 above)
Comment (liviuconcioiu) [reply regexes/client/browsers.yml]: Yes, is based on the regex for Yandex Browser.

File: regexes/client/browsers.yml (hunk 2 above)
Comment (liviuconcioiu) [reply regexes/client/browsers.yml]: Yes, is based on the regex for Yandex Browser.

**Comment by sgiehl** [comment]
Does it make sense to add detection for Alpha and Beta releases as a separate browser?
Imho those are prerelease versions of Yandex browser and no "new" browsers :thinking:

**Comment by liviuconcioiu** [comment]
It depends on how device-detector is used. For analytics point of view, it doesn't matter so much, but if it's used in affiliate marketing for example, it matters. It can differentiate tech guys (as most of the alpha, beta versions are used by them) for example from other users.

File: regexes/client/browsers.yml (hunk 3 above)
Comment (mimmi20) [reply regexes/client/browsers.yml]: Should the regex not be `'YaBrowser(?:/(\d+[\.\d]*))? \(alpha\)'` ? The last `?` in the Yandox browser regex makes the version part optional.

**Comment by sgiehl** [comment]
We can for sure include `alpha` or `beta` in the version. But imho not in the browser name.
e.g. use something like:

```
regex: YaBrowser(?:/(\d+[\.\d]*)) ?\((alpha|beta)\)?
version: $1 $2
```

That should result in versions like `17.4.0.491.00 alpha`

File: regexes/client/browsers.yml (hunk 2 above)
Comment (sgiehl) [reply regexes/client/browsers.yml]:
```suggestion
- regex: 'YaBrowser(?:/(\d+[\.\d]*)) \(lite\)'
```

The `?` in the end is indeed kind of useless. Or might it be possible the the user agent only contains `(lite` (without the closing `)`)

**Comment by sgiehl** [comment]
@liviuconcioiu could you apply the suggested changes and resolve the conflict? Would be good to merge afterwards.

**Pull request closed without merging**